openedx/core/djangolib/tests/test_markup.py

```diff
@@ -100,3 +101,69 @@ def test_strip_all_tags_but_br_returns_html(self):
         html = strip_all_tags_but_br('{name}<br><script>')
         html = html.format(name='Rock & Roll')
         self.assertEqual(html.decode(), u'Rock &amp; Roll<br>')
+
+    def test_clean_dengers_html_filter(self):
+        """ Verify filter removes expected tags """
+        template = Template(
+            u"""
+                <%page expression_filter="h"/>
+                <%!
+                from openedx.core.djangolib.markup import clean_dangers_html
+                %>
+                
+                <%
+                    html_content = '''
+                        <html>
+                            <head>
+                                <script type="text/javascript" src="evil-site"></script>
+                                <link rel="alternate" type="text/rss" src="evil-rss">
+                                <style>
+                                    body {
+                                        background-image: url(javascript:do_evil)
+                                    };
+                                    div {
+                                        color: expression(evil)
+                                    };
+                                </style>
+                            </head>
+                            <body onload="evil_function()">
```

Comment:
it appears test HTML template has some unnecessary tags/attrs i.e  `href="javascript:evil_function()`, `onload="evil_function()"`, `blink`.  Our test case is not relevant to those can we remove theme?

Reply:
Our filter is also removing unfamiliar HTML tags (e.g: `blink`) and javascript calls from attrs (e.g: `href="javascript:evil_function()` ). We are also asserting these checks in the test case. 
That is why I think these tags and attributes are necessary.